File: ds assignment 3/q5 assignment 5.py
def evaluate_postfix(expression):
    stack = []

    for char in expression:
        if char.isdigit():
            stack.append(int(char))
        else:
            b = stack.pop()
            a = stack.pop()
            if char == '+':
                stack.append(a + b)
            elif char == '-':
                stack.append(a - b)
            elif char == '*':
                stack.append(a * b)
            elif char == '/':
                stack.append(a // b)  # Integer division

    return stack.pop()


# Driver Code
expr = input("Enter postfix expression (single digit numbers): ")
print("Evaluation Result:", evaluate_postfix(expr))

File: ds assignment 3/q5 assignment 5.cpp
// Assignment 3 Q5.cpp
#include <iostream>
#include <stack>
#include <string>
#include <cmath>
using namespace std;

int evaluatePostfix(string exp) {
    stack<int> s;
    for (char c : exp) {
        if (isdigit(c))
            s.push(c - '0');
        else {
            int val2 = s.top(); s.pop();
            int val1 = s.top(); s.pop();
            switch (c) {
                case '+': s.push(val1 + val2); break;
                case '-': s.push(val1 - val2); break;
                case '*': s.push(val1 * val2); break;
                case '/': s.push(val1 / val2); break;
                case '^': s.push(pow(val1, val2)); break;
            }
        }
    }
    return s.top();
}

int main() {
    string postfix;
    cout << "Enter postfix expression: ";
    cin >> postfix;
    cout << "Evaluated result: " << evaluatePostfix(postfix) << endl;
    return 0;
}

